// Copyright 2017 The Chromium OS Authors. All rights reserved.
// Use of this source code is governed by a BSD-style license that can be
// found in the LICENSE file.

syntax = "proto3";

package puffin;
option optimize_for = LITE_RUNTIME;

message ProtoByteExtent {
  uint64 offset = 1;
  uint64 length = 2;
}

message StreamInfo {
  repeated ProtoByteExtent deflates = 1;
  repeated ProtoByteExtent puffs = 2;
  uint64 puff_length = 3;
}

message PatchHeader {
  int32 version = 1;
  StreamInfo src = 2;
  StreamInfo dst = 3;
  // The bsdiff patch is installed right after this protobuf.
}